index.js

```diff
@@ -22,10 +23,16 @@ module.exports = function (opts) {
 			from: file.path,
 			to: file.path
 		}).then(function (res) {
+			var map;
 			file.contents = new Buffer(res.css);
 
 			if (res.map && file.sourceMap) {
-				applySourceMap(file, res.map.toString());
+				map = res.map.toJSON();
+				map.file = file.relative;
+				map.sources = [].map.call(map.sources, function (source) {
+					return path.join(path.dirname(file.relative), source);
+				});
```

Comment:
No, I mean the whole change. Why is any of this needed. It should just work. I don't think it makes sense to work around stuff here. It should rather be fixed upstream so it benefits all consumers.